### ISSUE TITLE
`MemoryReader.timeseries` doesn't raise NoDataError on empty AtomGroup, while `DCDReader.timeseries` does

### ISSUE THREAD

**Issue by hmacdope** [MDAnalysis/mdanalysis#3898]
## Expected behavior ##

The DCDReader timeseries and MemoryReader behave the same.

## Actual behavior ##

The `DCDReader` raises a `NoDataError` if an empty atomgroup is provided as the `asel` argument. This is the same approach taken in #3890 

However the `MemoryReader` just returns an empty array.

## Code to reproduce the behavior ##

``` python
import MDAnalysis as mda
from MDAnalysis.tests.datafiles import PSF, DCD,
import numpy as np

u = mda.Universe(PSF, DCD)
u.trajectory.timeseries(u.select_atoms(None)) # raises NoDataError

coords = np.zeros((10,20,3))
u = mda.Universe(coords, format="MEMORY")
u.trajectory.timeseries(u.select_atoms(None)) # returns an empty array. 
```

**Comment by IAlibay** [comment]
I believe the recent consensus was that NoDataError was for missing topology attributes rather than zero sized things (see https://github.com/MDAnalysis/mdanalysis/issues/3837#issuecomment-1256617306)

Probably would want some other error type here?

**Comment by hmacdope** [comment]
We could go for ValueError or something, whatever people think is best. Is this best addressed in a seperate PR after #3890 where we homogenise all the call sites?

**Comment by IAlibay** [comment]
> We could go for ValueError or something, whatever people think is best. Is this best addressed in a seperate PR after #3890 where we homogenise all the call sites?

I probably would just advocate for changing DCD's to ValueError and make the other two ValueError? Seems like a simple enough one to have within the scope of #3890?

**Comment by orbeckst** [comment]
I agree with moving to `ValueError`.

Thanks for raising the discussion @IAlibay , making the semantics of a historical code base consistent is hard work.

**Comment by IAlibay** [comment]
We really should document this somewhere, i.e. "Raised when empty input is not allowed or required data are missing." is very non-specific if we're going with a TopologyAttrs centric viewpoint. I'll raise an issue.

**Comment by hmacdope** [comment]
Fixed in #3890
